### PR TITLE
TASK: Improve indexing errors handling

### PR DESCRIPTION
**Pull request by dfeyer**
This change create a file, like the SystemLogger, stored in
```Data/Logs/ElasticSearch/``` with the current bulk request and the
errors returned by ElasticSearch in pretty formatted JSON.

### CODE REVIEW

**Comment by dfeyer** [comment]
:+1:

File: Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Indexer/NodeIndexer.php [hunk 1]
```diff
@@ -398,6 +398,35 @@ public function flush() {
...
+	 * @param string $errors
+	 */
+	protected function logIndexingErrors($bulkRequest, $errors) {
+		if (!file_exists(FLOW_PATH_DATA . 'Logs/ElasticSearch')) {
+			mkdir(FLOW_PATH_DATA . 'Logs/ElasticSearch');
+		}
+		if (file_exists(FLOW_PATH_DATA . 'Logs/ElasticSearch') && is_dir(FLOW_PATH_DATA . 'Logs/ElasticSearch') && is_writable(FLOW_PATH_DATA . 'Logs/ElasticSearch')) {
+			$referenceCode = date('YmdHis', $_SERVER['REQUEST_TIME']) . substr(md5(rand()), 0, 6);
+			$dumpPathAndFilename = FLOW_PATH_DATA . 'Logs/ElasticSearch/' . $referenceCode . '.txt';
+			file_put_contents($dumpPathAndFilename, $this->renderIndexingErrors($bulkRequest, $errors));
```
Comment (aertmann) [inline Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Indexer/NodeIndexer.php]: I highly discourage this custom approach. Instead use the proper logging API, see an example here https://github.com/mocdk/MOC.Varnish/commit/06f8b8f49ef34bc28a0c74fab677d6d80a024e81

**Comment by aertmann** [comment]
Very helpful, always hated debugging those log entries.

**Comment by aertmann** [comment]
Also could you provide an example of the before/after?

**Comment by dfeyer** [comment]
Close this one in favor of https://github.com/Flowpack/Flowpack.ElasticSearch.ContentRepositoryAdaptor/pull/151

**Pull request closed without merging**